### ISSUE TITLE
downloadable font: rejected by sanitizer /fonts/fontawesome-webfont.woff2?v=4.6.3

### ISSUE THREAD

**Issue by jujes** [reactioncommerce/reaction#1134]
### General question   

**Development branch build of reactioncommerce:reaction**

Someone know about this font issue?

![screen shot 2016-07-05 at 11 46 46 pm](https://cloud.githubusercontent.com/assets/2881017/16601066/c7d8f5f2-430a-11e6-94fd-e4b9c020d439.png)

```
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:23956
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:24274
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:23956
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:24274
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create f474c6bbefe48578647e20819f049973ea16351c.js:26:20071
downloadable font: rejected by sanitizer (font-family: "FontAwesome" style:normal weight:normal stretch:normal src index:1) source: https://rootsoap.com/fonts/fontawesome-webfont.woff2?v=4.6.3 /:6985:12
downloadable font: rejected by sanitizer (font-family: "FontAwesome" style:normal weight:normal stretch:normal src index:2) source: https://rootsoap.com/fonts/fontawesome-webfont.woff?v=4.6.3 /:6985:12
downloadable font: rejected by sanitizer (font-family: "FontAwesome" style:normal weight:normal stretch:normal src index:3) source: https://rootsoap.com/fonts/fontawesome-webfont.ttf?v=4.6.3 /:6985:12
```

Thanks in advance,

**Comment by aaronjudd** [comment]
Prequel is the `development` branch.   Need to look into that error though, that's the first I've seen of it. @jshimko ideas?

**Comment by jshimko** [comment]
I can't reproduce this, but [a quick Google search](https://www.google.com/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=downloadable%20font%20rejected%20by%20sanitizer) turns up a few common causes for that error with Firefox.

[Memory issues or a corrupt font file](https://support.mozilla.org/en-US/questions/913498)...
> You get this error if you run out of memory when loading the fontfile or if there is something wrong with the layout (contents) of the fontfile. This is a protection against bad or malicious font files. 

[The query string in the font path](http://jamie.curle.io/posts/firefox-font-rejected-by-sanitzer/)
(that guy is getting the query string added by a different preprocessor, but we [already have them by default] (https://github.com/reactioncommerce/reaction/blob/development/imports/themes/default/styles/fonts.less#L8-L9) before it's even processed)

[Single quotes instead of double quotes](http://stackoverflow.com/questions/11072655/firefox-font-face-fail-with-fontawesome) (seriously Firefox??)
 
Not sure if it's any of these though.  As I said, I can't reproduce this.  What version of Firefox are you using?  Does it happen for you here?  https://hello.reactioncommerce.com/ (that's the development branch as of today)

**Comment by jujes** [comment]
@aaronjudd @jshimko also in chrome have the same issue:

my browser **version**:
**chrome**: 51.0.2704.106 (64-bit)
**firefox**: 47.0.1

I user jwilder/nginx-proxy + custom

```
FROM jwilder/nginx-proxy:latest

RUN { \
      echo 'fastcgi_buffers 4 256k;'; \
      echo 'fastcgi_buffer_size 128k;'; \
      echo 'fastcgi_busy_buffers_size 256k;'; \
    } > /etc/nginx/conf.d/my_proxy.conf

RUN sed -i 's/^http {/&\n    proxy_busy_buffers_size   256k;/g' /etc/nginx/nginx.conf && \
    sed -i 's/^http {/&\n    proxy_buffers   4 256k;/g' /etc/nginx/nginx.conf && \
    sed -i 's/^http {/&\n    proxy_buffer_size   128k;/g' /etc/nginx/nginx.conf
```

![screen shot 2016-07-06 at 1 47 19 pm](https://cloud.githubusercontent.com/assets/2881017/16616870/63e19c36-4380-11e6-96fe-fa07c2244a8e.png)

```
Navigated to https://rootsoap.com/
(index):1 Failed to decode downloaded font: https://rootsoap.com/fonts/fontawesome-webfont.woff2?v=4.6.3
(index):1 OTS parsing error: invalid version tag
(index):1 Failed to decode downloaded font: https://rootsoap.com/fonts/fontawesome-webfont.woff2?v=4.6.3
(index):1 OTS parsing error: invalid version tag
(index):1 Failed to decode downloaded font: https://rootsoap.com/fonts/fontawesome-webfont.woff?v=4.6.3
(index):1 OTS parsing error: invalid version tag
(index):1 Failed to decode downloaded font: https://rootsoap.com/fonts/fontawesome-webfont.woff?v=4.6.3
(index):1 OTS parsing error: invalid version tag
(index):1 Failed to decode downloaded font: https://rootsoap.com/fonts/fontawesome-webfont.ttf?v=4.6.3
(index):1 OTS parsing error: invalid version tag
(index):1 Failed to decode downloaded font: https://rootsoap.com/fonts/fontawesome-webfont.ttf?v=4.6.3
(index):1 OTS parsing error: invalid version tag
```

**Comment by aaronjudd** [comment]
Right now, the font awesome fonts are being copied by the npm install script from the npm package into the public/fonts directory.  This must not be happening.

**Comment by jujes** [comment]
10MB diference between **prequel** and my **reaction-dev**
if I use prequel


```
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:23956
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:24274
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:23956
unreachable code after return statement f474c6bbefe48578647e20819f049973ea16351c.js:299:24274
mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create f474c6bbefe48578647e20819f049973ea16351c.js:26:20071
```

### docker ps -a

```
reactioncommerce/prequel   latest              e6fcaa78d335        22 hours ago        881 MB
jujes/reaction-dev         latest              700ad230a52a        17 hours ago        879.6 MB 
```

**Comment by jshimko** [comment]
@jujes Does the problem happen here?  https://hello.reactioncommerce.com/

**Comment by jujes** [comment]
@jshimko no, here capture about https://hello.reactioncommerce.com/
![screen shot 2016-07-06 at 3 33 10 pm](https://cloud.githubusercontent.com/assets/2881017/16619600/19b9e014-438f-11e6-89d3-2aaf198bdf64.png)

**Comment by jshimko** [comment]
I can't even reproduce this on _your_ site.

![fa-firefox](https://cloud.githubusercontent.com/assets/3077956/16621335/d9953d60-4363-11e6-9261-5dcb780c5e66.jpg)

![fa-chrome](https://cloud.githubusercontent.com/assets/3077956/16621333/d98e181e-4363-11e6-9246-3778ee4a5be9.jpg)

![fa-chrome2](https://cloud.githubusercontent.com/assets/3077956/16621334/d9934c08-4363-11e6-822b-b84ad438e27c.jpg)

**Comment by jujes** [comment]
@jshimko the site what you are looking **rootsoap.com** run reactioncommerce/prequel
for develop propose I modified /etc/hosts and add this domain to build and deploy locally...

Now I move the development scripts online to look if you can see the issue... 

please check: rootsoap.com

**Comment by jshimko** [comment]
Yep.  I get the error now.  Only on your site though.  I still don't get it anywhere else.

**Comment by jujes** [comment]
@jshimko 
I build it again and get the same issue in **three different machines** (VPS, VirtualBox, BareMetal)
here the terminal error output when I build:

```
debconf: delaying package configuration, since apt-utils is not installed
Error checking path, continuing { [Error: Cannot find module '/usr/lib/node_modules/phantomjs-prebuilt/lib/location'] code: 'MODULE_NOT_FOUND' }
npm WARN lifecycle reaction@0.14.0~postinstall: cannot run in wd %s %s (wd=%s) reaction@0.14.0 cp -R node_modules/font-awesome/fonts ./public/ /var/src
npm WARN lifecycle meteor-dev-bundle@0.0.0~install: cannot run in wd %s %s (wd=%s) meteor-dev-bundle@0.0.0 node npm-rebuild.js /var/www/bundle/programs/server
npm WARN meteor-dev-bundle@0.0.0 No description
npm WARN meteor-dev-bundle@0.0.0 No repository field.
npm WARN meteor-dev-bundle@0.0.0 No license field.
```

Regards,

**Comment by jujes** [comment]
Comparing reactioncommerce/prequel and my deploy container have this differences:

```
top
  hello.reactioncommerce.com
    assets/files/Media
    fonts
    index
    .
    .

top
  rootsoap.com
    resources
    index (issues)
      2 Failed to decode downloaded font: http://rootsoap.com/fonts/fontawesome-webfont.woff2?v=4.6.3
      6 OTS parsing error: invalid version tag
      2 Failed to decode downloaded font: http://rootsoap.com/fonts/fontawesome-webfont.woff?v=4.6.3
      2 Failed to decode downloaded font: http://rootsoap.com/fonts/fontawesome-webfont.ttf?v=4.6.3
    .
    .
```

![screen shot 2016-07-14 at 1 14 52 pm](https://cloud.githubusercontent.com/assets/2881017/16837811/ea2083ca-49c5-11e6-8148-3334438db0a9.png)

![screen shot 2016-07-14 at 1 15 17 pm](https://cloud.githubusercontent.com/assets/2881017/16837819/efa3ade0-49c5-11e6-8138-307c238a91ac.png)

![screen shot 2016-07-14 at 1 15 39 pm](https://cloud.githubusercontent.com/assets/2881017/16837825/f6f7fd8a-49c5-11e6-8022-7e2201f2002b.png)

**Comment by hrath2015** [comment]
Try running this command `cp -R node_modules/font-awesome/fonts ./public/` from install directory.
See if it helps.

**Comment by jujes** [comment]
**thanks** @hrath2015 my solution come from rebuild the container following your advice..

- get [Font-Awesome](https://github.com/FortAwesome/Font-Awesome/tree/master/fonts) master fonts into /public/resources/fonts

```
FontAwesome.otf
fontawesome-webfont.eot
fontawesome-webfont.svg
fontawesome-webfont.ttf
fontawesome-webfont.woff
fontawesome-webfont.woff2
```
- change permissions

```
chmod -R 775 /public/resources/fonts
```

- **remap** imports/themes/default/styles/fonts.less **adding** /resources/ path and **removing** ?v=4.6.3

```
@font-face {
  font-family: 'FontAwesome';
  src: url('/resources/fonts/fontawesome-webfont.eot');
  src: url('/resources/fonts/fontawesome-webfont.eot') format('embedded-opentype'), url('/resources/fonts/fontawesome-webfont.woff2') format('woff2'), url('/resources/fonts/fontawesome-webfont.woff') format('woff'), url('/resources/fonts/fontawesome-webfont.ttf') format('truetype'), url('/resources/fonts/fontawesome-webfont.svg#fontawesomeregular') format('svg');
  font-weight: normal;
  font-style: normal;
}
```

If someone know a better way to solve this issue, please let me know.

**Comment by aaronjudd** [comment]
The command @hrath2015 mentioned `cp -R node_modules/font-awesome/fonts ./public/ ` is being run by npm in the `post-install` step, when "meteor npm install` is performed in development. All I can think is that you're doing a `meteor build` without actually doing a `meteor npm install` first?  You should not have to change the less..  although that's ok, the font awesome fonts are just being put in /public/fonts (not public/resources/fonts which is just where we put the reaction logo which is a font) .. maybe we should consolidate and pick one though..  
Note that meteor strips off the "public" so the dir path in less should just be "/fonts" if you copy to `public/fonts`.

**Comment by jujes** [comment]
@aaronjudd **true** in order with your explanation all what I need is put [Font-Awesome](https://github.com/FortAwesome/Font-Awesome/tree/master/fonts) in /public/fonts

- `chmod -R 775 /public/fonts`
- `docker build -t own/prequel-dev .`

**Comment by jshimko** [comment]
Assuming you used the standard Dockerfile (which your command implies), you don't need to manually copy over the fonts.  It already does it for you [when it runs `meteor npm install`](https://github.com/reactioncommerce/reaction/blob/development/docker/scripts/build-meteor.sh#L20) because the [package.json runs this command](https://github.com/reactioncommerce/reaction/blob/development/package.json#L22) every time you install NPM modules:
```sh
cp -R node_modules/font-awesome/fonts ./public/
```
I can confirm has been working too.  Aside from checking inside a freshly built container to make sure the fonts directory is there, I also noticed your site had FontAwesome loading (see my images posted above).  The only way that could work is if those font files were successfully moved over to the public folder.  So I suspect there's something more to the problem.  But since I can't reproduce it anywhere else, I'm inclined to think it's something with your particular build.

**Comment by ingenial** [comment]
I que the same error in firefox but not in chrome 
file:///root/Pictures/Selection_007.png

**Comment by jshimko** [comment]
This has been fixed by #1223.
